ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/Photo.java

```diff
@@ -29,43 +29,77 @@
     private byte[] mJpeg;
     private Exif.RequiredTags mRequiredTags;
     private int mRotationForDisplay = 0;
-    private final String mUUID;
+    private String mUUID = "";
     private int mRotationDelta = 0;
 
     private PhotoEdit mEditor;
 
-    public static Photo fromJpeg(@NonNull byte[] jpeg, int orientation) {
-        Photo photo = new Photo();
-        photo.setJpeg(jpeg);
-        photo.setBitmapPreview(createPreview(jpeg));
-        photo.setRotationForDisplay(orientation);
-        photo.readRequiredTags();
-        photo.updateExif();
-        return photo;
+    public static Photo fromJpeg(@NonNull final byte[] jpeg, final int orientation) {
+        return new Photo(jpeg, orientation);
```

Comment:
Refactored this, because the steps done in this factory method are required for the construction of a Photo instance so they should go into the constructor.